extensions/sovity-messenger/src/test/java/de/sovity/edc/extension/messenger/demo/SovityMessengerDemoTest.java

```diff
@@ -0,0 +1,133 @@
+        emitterEdcContext.setConfiguration(providerConfig.getProperties());
+        emitterEdcContext.registerServiceMock(TokenDecorator.class, (td) -> td);
+
+        consumerConfig = forTestDatabase(RECEIVER_PARTICIPANT_ID, RECEIVER_DATABASE);
+        receiverEdcContext.setConfiguration(consumerConfig.getProperties());
+        receiverEdcContext.registerServiceMock(TokenDecorator.class, (td) -> td);
+
+        receiverAddress = "http://localhost:" + consumerConfig.getProtocolEndpoint().port() + consumerConfig.getProtocolEndpoint().path();
+    }
+
+    /**
+     * Actual usage of the Sovity Messenger.
+     */
+    @DisabledOnGithub
+    @Test
+    void demo() throws ExecutionException, InterruptedException, TimeoutException {
+        /*
+         * Get a reference to the SovityMessenger. This is equivalent to
+         *
+         * @Inject SovityMessenger messenger;
+         *
+         * in an extension.
+         *
+         * This messenger is already configured to accept messages in de.sovity.edc.extension.messenger.demo.SovityMessengerDemo#initialize
+         */
+        val messenger = emitterEdcContext.getContext().getService(SovityMessenger.class);
+
+        System.out.println("START MARKER");
+
+        // Send messages
+        val added = messenger.send(Answer.class, receiverAddress, new Addition(20, 30));
+        val rooted = messenger.send(Answer.class, receiverAddress, new Sqrt(9.0));
+        val unregistered = messenger.send(Answer.class, receiverAddress, new UnregisteredMessage());
+        messenger.send(receiverAddress, new Signal());
+
+        try {
+            // Wait for the answers
+            added.get(2, TimeUnit.SECONDS).onSuccess(it -> System.out.println(it.getAnswer()));
+            rooted.get(2, TimeUnit.SECONDS).onSuccess(it -> System.out.println(it.getAnswer()));
+            unregistered.get(2, TimeUnit.SECONDS);
+        } catch (ExecutionException e) {
+            /*
+             * When a problem happens, a SovityMessengerException is thrown and encapsulated in an ExecutionException.
+             */
+            System.out.println(e.getCause().getMessage());
+        }
+
+        try {
+            val failing1 = messenger.send(Answer.class, receiverAddress, new Failing("Some content 1"));
+            val failing2 = messenger.send(Answer.class, receiverAddress, new Failing("Some content 2"));
+            failing1.get(2, TimeUnit.SECONDS);
+            failing2.get(2, TimeUnit.SECONDS);
+        } catch (ExecutionException e) {
+            val cause = e.getCause();
+            if (cause instanceof SovityMessengerException messengerException) {
+                // Error when processing a message with type demo-failing
+                System.out.println(messengerException.getMessage());
+                // {"message":"Some content 1/2"}
+                System.out.println(messengerException.getBody());
+            }
+        }
```

Comment:
assertThatThrownBy(() -> {


}).isInstanceOf(...).hasMessage(...

Reply:
No, this is the demo, this is to show how the code would be used in a real usecase.